pkg/command/org_validation_helper.go

```diff
@@ -0,0 +1,82 @@
+// Copyright 2024 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package command
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/pkg/errors"
+
+	"github.com/vmware-tanzu/tanzu-plugin-runtime/config"
+	"github.com/vmware-tanzu/tanzu-plugin-runtime/log"
+
+	"github.com/vmware-tanzu/tanzu-cli/pkg/auth/csp"
+	"github.com/vmware-tanzu/tanzu-cli/pkg/centralconfig"
+	cliconfig "github.com/vmware-tanzu/tanzu-cli/pkg/config"
+)
+
+const centralConfigTanzuApplicationPlatformScopesKey = "cli.core.tanzu_application_platform_scopes"
+
+type tapScope struct {
+	Scope string `yaml:"scope" json:"scope"`
+}
+
+type tapScopesGetter func() ([]string, error)
+
+// validateTokenForTAPScopes validates if the claims contain required TAP scopes.
+func validateTokenForTAPScopes(claims *csp.Claims, scopesGetter tapScopesGetter) (bool, error) {
+	if scopesGetter == nil {
+		scopesGetter = getTAPScopesFromCentralConfig
```

Comment:
nit: Do we need worry about insisting on a single set of scopes to be used by all tanzu contexts? 
at minimum should we provide a CLI option to disable the scope validation?

Reply:
Thank you for pointing it out. Provided an environment variable (`TANZU_CLI_SKIP_TAP_SCOPES_VALIDATION_ON_TANZU_CONTEXT`) to skip the scope validation on tanzu contexts